application/src/main/java/org/togetherjava/tjbot/commands/free/ChannelStatus.java

```diff
@@ -0,0 +1,205 @@
+        } else
+            throw new IllegalStateException(
+                    "This channel status was created with the id for a non-text-channel and status cannot be monitored: '%s'"
+                        .formatted(channelId));
+    }
+
+    /**
+     * Method to set the channel status to busy, a user id is passed in to keep track of the current
+     * user requesting help. This id will be used to confirm that the author is satisfied with the
+     * channel being marked as free.
+     * <p>
+     * This functionality is not yet implemented so the id can be anything atm. Also note that on
+     * reboot the bot does not currently search for the author so the first time its marked as free
+     * there will be no confirmation.
+     * 
+     * @param userId the id of the user who changed the status to 'busy'
+     */
+    public void setBusy(final long userId) {
+        if (isBusy == FREE) {
+            this.isBusy = BUSY;
+            this.userId = userId;
+        }
+    }
+
+    /**
+     * Method to set the channel status to free, the user id of the previous help requester is not
+     * overwritten by this method. So until another user changes the status to busy the old value
+     * will remain.
+     * <p>
+     * The value will be 0 until the first time that the status is changed from free to busy.
+     * <p>
+     * This functionality is not yet implemented so the id can be anything atm.
+     */
+    public void setFree() {
+        if (isBusy == BUSY) {
+            isBusy = FREE;
+        }
+    }
+
+    // todo should I overload equals with equals(long) so that a Set may be used instead of a Map
+    /**
+     * The identity of this object of solely based on the id value. Compares the long id's and
+     * determines if they are equal.
+     * 
+     * @param o the other object to test against
+     * @return whether the objects have the same id or not.
+     */
+    @Override
+    public boolean equals(final Object o) {
+        if (this == o)
+            return true;
+        if (o == null || getClass() != o.getClass())
+            return false;
+
+        ChannelStatus status = (ChannelStatus) o;
+        return channelId == status.channelId;
+    }
+
+    /**
+     * A String representation of the instance, gives the name and the current status.
+     *
+     * @return a String representation of the instance.
+     */
+    @Override
+    public @NotNull String toString() {
+        return "ChannelStatus{ %s is %s }".formatted(name, isBusy ? "busy" : "not busy");
+    }
+
+    /**
+     * A {@link #toString()} method specially formatted for Discord ({@link JDA}. Uses emojis by
+     * string representation, that discord will automatically convert into images. Using this string
+     * outside of discord will display unexpected results.
+     *
+     * @return a String representation of ChannelStatus, formatted for Discord
+     */
+    public @NotNull String toDiscord() {
```

Comment:
i mentioned somewhere earlier that the name of this method is not ideal and could be renamed to sth like `toDiscordMessage` or similar.

Reply:
fixed